app/backend/approaches/chatreadretrieveread.py

```diff
@@ -96,106 +96,89 @@ async def run_until_final_call(
         top = overrides.get("top", 3)
         minimum_search_score = overrides.get("minimum_search_score", 0.0)
         minimum_reranker_score = overrides.get("minimum_reranker_score", 0.0)
+        include_category = overrides.get("include_category")
         filter = self.build_filter(overrides, auth_claims)
 
         original_user_query = messages[-1]["content"]
         if not isinstance(original_user_query, str):
             raise ValueError("The most recent message content must be a string.")
         user_query_request = "Generate search query for: " + original_user_query
 
-        tools: List[ChatCompletionToolParam] = [
-            {
-                "type": "function",
-                "function": {
-                    "name": "search_sources",
-                    "description": "Retrieve sources from the Azure AI Search index",
-                    "parameters": {
-                        "type": "object",
-                        "properties": {
-                            "search_query": {
-                                "type": "string",
-                                "description": "Query string to retrieve documents from azure search eg: 'Health care plan'",
-                            }
+        sources_content = []
+        extra_info = {"thoughts": [], 'data_points': []}
+
+        if include_category != "__NONE__":
```

Comment:
I think `include_category != "__NONE__"` condition should be executed only once, stored in a variable, and be under a comment that says that __NONE__ is a value passed down by the frontend categories picker. 
I'm nervous about special magical values like __NONE__ so want to make it clear where it came from and minimize places to use it incorrectly.
It could also be a class attribute on Approach, like NO_CATEGORIES